### PR TITLE
feat: add trending topics Trigger.dev scheduled task

### LINKED ISSUE #192: feat: Build trending topics Trigger.dev scheduled task
Part of #185

## Context

The dashboard needs a "Trending Topics" headline showing AI-extracted topic clusters from recent episode summaries. This requires a scheduled background job that periodically analyzes summaries and caches results.

## Current State

- `src/trigger/` — Existing Trigger.dev tasks (summarize-episode, import-opml, send-notification-digests)
- `src/lib/ai/` — AI provider abstraction layer (OpenRouter + Z.AI) — ADR-008
- `src/db/schema.ts` — Episodes have `summary`, `keyTakeaways` fields populated by summarization
- No trending topics infrastructure exists

## Approach

1. **New DB table** `trending_topics` in `src/db/schema.ts`:
   - `id` (serial PK)
   - `topics` (json) — Array of `{ topic: string, episodeCount: number, sampleEpisodeIds: number[] }`
   - `generatedAt` (timestamp) — When this snapshot was created
   - `periodStart` (timestamp) — Start of the analysis window
   - `periodEnd` (timestamp) — End of the analysis window
2. **New Trigger.dev scheduled task** `src/trigger/generate-trending-topics.ts`:
   - Runs daily (or every 6 hours — configurable via cron)
   - Query: SELECT summaries + key_takeaways from episodes published in last 7 days that have been summarized
   - Send batch to LLM via AI provider abstraction: "Extract 5-8 trending topics from these episode summaries. For each topic, provide a short label and count of episodes discussing it."
   - Parse LLM response, store in `trending_topics` table
   - Keep last N snapshots (e.g., 7 days) for potential trend-over-time display later
3. **New server action** `getTrendingTopics()` in `src/app/actions/dashboard.ts`:
   - Fetch most recent `trending_topics` row
   - Return parsed topics array
   - Cache-friendly: same result until next job run
4. **Generate migration**: `bun run db:generate` → `bun run db:push`

## Files to Modify

- `src/db/schema.ts` — Add `trendingTopics` table + types
- `src/trigger/generate-trending-topics.ts` (new) — Scheduled Trigger.dev task
- `src/app/actions/dashboard.ts` — Add `getTrendingTopics()` server action
- `src/lib/prompts.ts` — Add trending topics extraction prompt template
- `drizzle/` — Generated migration files

## Acceptance Criteria

- VERIFY: `trending_topics` table exists with columns: id, topics (json), generatedAt, periodStart, periodEnd
- VERIFY: Trigger.dev task runs on schedule and populates the table
- VERIFY: LLM prompt extracts 5-8 distinct topic clusters with episode counts
- VERIFY: `getTrendingTopics()` returns the most recent snapshot
- VERIFY: Task uses AI provider abstraction (not hardcoded to one provider)
- VERIFY: Task handles edge case: no summarized episodes in window → stores empty topics array
- VERIFY: Old snapshots are retained (not deleted) for potential future use
- VERIFY: Migration runs cleanly via `bun run db:push`

## Complexity Hint

- Subsystems: database schema, Trigger.dev tasks, AI provider, server actions
- Structural changes: yes (new table + scheduled task)

## Out of Scope

- Trend-over-time visualization (showing how topics change)
- User-specific topic preferences or filtering
- Real-time topic updates
- Topic-based episode search/filtering

### PR DESCRIPTION
**Pull request by funny-otter**
## Summary

- Add daily scheduled Trigger.dev task (`generate-trending-topics`) that queries recently summarized episodes, extracts 5-8 topic clusters via LLM, and stores append-only snapshots in a new `trending_topics` table
- Add `getTrendingTopics()` server action for dashboard consumption (returns most recent snapshot)
- Add `TRENDING_TOPICS_SYSTEM_PROMPT` and `getTrendingTopicsPrompt()` prompt templates using the AI provider abstraction
- Add ADR-022 documenting the daily LLM snapshot approach vs alternatives
- Handle edge cases: empty episode window, missing takeaways, LLM parse failure, hallucinated episode IDs

Closes #192

## Test plan

- [x] Unit tests for trigger task (7 tests): empty window, no takeaways, happy path, parse failure, invalid IDs, zero-valid-after-filter, count correction
- [x] Unit tests for server action (4 tests): unauthenticated, no snapshots, latest snapshot, DB error
- [x] `bun run lint` — passes
- [x] `bun run test` — 1124 tests pass (101 files)
- [x] `bun run build` — compiles successfully
- [ ] Run `bun run db:push` against dev database to verify migration
- [ ] Manual: trigger task via Trigger.dev dashboard, verify row in `trending_topics`
- [ ] Production: run `doppler run --config prd -- bunx drizzle-kit push` after merge

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Trending topics on the dashboard: daily 5–8 topic clusters with descriptions, episode counts, time window (auto-updated snapshot); dashboard can fetch the latest snapshot.
  * Listen history tracking to record episode play attempts and progress.

* **Documentation**
  * Added design doc detailing the daily trending snapshot approach, retention, token budget, and empty-window behavior.

* **Tests**
  * Added unit and end-to-end tests for trending topic generation and dashboard retrieval.

* **Chores**
  * Database migrations to support trending snapshots, listen history, and rate limiting.

### CODE REVIEW

File: src/app/actions/dashboard.ts [hunk 1]
```diff
@@ -212,6 +217,39 @@ export async function getDashboardStats() {
...
+  const { userId } = await auth();
+
+  if (!userId) {
+    return { topics: null, error: "You must be signed in" };
+  }
+
+  try {
+    const latest = await db.query.trendingTopics.findFirst({
+      orderBy: [desc(trendingTopics.generatedAt)],
+    });
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/app/actions/dashboard.ts]: Fixed: added desc(trendingTopics.id) as tiebreaker to orderBy

File: src/trigger/generate-trending-topics.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        const filteredIds = topic.episodeIds.filter((id) =>
+          validEpisodeIds.has(id)
+        );
+        return {
+          ...topic,
+          episodeIds: filteredIds,
+          episodeCount: filteredIds.length,
+        };
+      })
+      .filter((topic) => topic.episodeCount > 0);
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/generate-trending-topics.ts]: Fixed: validation now checks description (string) and episodeIds elements (integers). Post-validation deduplicates IDs, sorts by episodeCount desc, and clamps to MAX_TOPICS=8

File: src/trigger/generate-trending-topics.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/generate-trending-topics.ts]: Fixed: added logger.warn when validatedTopics is empty but parsed topics existed, distinguishing hallucination from legitimate empty window

File: src/trigger/generate-trending-topics.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      .from(episodes)
+      .where(
+        and(
+          eq(episodes.summaryStatus, "completed"),
+          isNotNull(episodes.processedAt),
+          gte(episodes.processedAt, periodStart),
+          isNotNull(episodes.keyTakeaways)
+        )
+      )
+      .limit(MAX_EPISODES);
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/generate-trending-topics.ts]: Fixed: added .orderBy(desc(episodes.processedAt)) before .limit(MAX_EPISODES) for deterministic episode selection

File: src/trigger/generate-trending-topics.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      episodeCount: number
+    ) => {
+      await db.insert(trendingTopics).values({
+        topics,
+        generatedAt: now,
+        periodStart,
+        periodEnd,
+        episodeCount,
+      });
+    };
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/generate-trending-topics.ts]: Acknowledged — intentional design. `generatedAt` captures the generation run start time, not the row insertion time (that's what the schema's `createdAt` with `defaultNow()` is for). All snapshot timestamps (`periodStart`, `periodEnd`, `generatedAt`) use the same `now` reference for consistency within a single run. The delta from the LLM call is a few seconds at most and doesn't affect ordering or display.
